### PR TITLE
[random123] Add new port

### PR DESCRIPTION
**Pull request by KineticTheory**
**Provide port for Deshaw Research's Random123 implementation**

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  `All` / `Yes` - This is a header only package.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  `Yes`

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/random123/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "name": "random123",
+  "version-string": "1.14.0",
```
Comment (Thomas1664) [inline ports/random123/vcpkg.json:3]:
```suggestion
  "version": "1.14.0",
```

File: ports/random123/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    ARCHIVE
+    URLS "https://github.com/DEShawResearch/random123/archive/refs/tags/v${VERSION}.tar.gz"
+    FILENAME "v${VERSION}.tar.gz "
+    SHA512 1c7d139193f5404e5d14d229f55e0a14b11de596a4cfbf0a39c1419f5ae146055dccc61e9430f724a2d7c1efb8bd01edb72866d4f4705508fcc6ebda6e90e91e
+)
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
+)
```
Comment (Thomas1664) [inline ports/random123/portfile.cmake:15]: Please use [vcpkg_from_github](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_from_github.md) instead.

Comment (KineticTheory) [reply ports/random123/portfile.cmake]: Thanks for the feedback.  Syntax corrected in [4a026fd](https://github.com/microsoft/vcpkg/pull/24716/commits/4a026fdc693bf155dd720816381786d39e2799e2)

**Comment by LilyWangLL** [comment]
Thanks for adding this port, please run command `vcpkg x-add-version random123 --overwrite-version` on vcpkg window.

File: ports/random123/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  SHA512 1c7d139193f5404e5d14d229f55e0a14b11de596a4cfbf0a39c1419f5ae146055dccc61e9430f724a2d7c1efb8bd01edb72866d4f4705508fcc6ebda6e90e91e
+  HEAD_REF "main"
+  )
+
+# Copy the headers that define this package to the install location.
+file(GLOB header_files 
+     ${SOURCE_PATH}/include/Random123/*.h 
+     ${SOURCE_PATH}/include/Random123/*.hpp ) 
+file(COPY ${header_files}
+     DESTINATION ${CURRENT_PACKAGES_DIR}/include/${PORT} )
```
Comment (LilyWangLL) [inline ports/random123/portfile.cmake:18]:
```suggestion
     DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}" )
```

File: ports/random123/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+file(COPY ${SOURCE_PATH}/include/Random123/conventional
+     DESTINATION ${CURRENT_PACKAGES_DIR}/include/${PORT} )
```
Comment (LilyWangLL) [inline ports/random123/portfile.cmake:20]:
```suggestion
     DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}" )
```

File: ports/random123/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+file(COPY ${SOURCE_PATH}/include/Random123/features
+     DESTINATION ${CURRENT_PACKAGES_DIR}/include/${PORT} )
```
Comment (LilyWangLL) [inline ports/random123/portfile.cmake:22]:
```suggestion
     DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}" )
```

File: ports/random123/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```
Comment (LilyWangLL) [inline ports/random123/portfile.cmake:25]:
```suggestion
file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by KineticTheory** [comment]
> Thanks for adding this port, please run command `vcpkg x-add-version random123 --overwrite-version` on vcpkg window.

I applied the fixes you requested.  Thank you for providing feedback.

**Comment by Thomas1664** [comment]
> I applied the fixes you requested. Thank you for providing feedback.

This didn't work because you modified the portfile without committing this change **before** running `x-add-version`. You have to run `vcpkg x-add-version random123 --overwrite-version` again.

**Comment by KineticTheory** [comment]
> vcpkg x-add-version random123 --overwrite-version

I didn't understand that they needed to be in separate commits.  I updated the version again.

**Comment by KineticTheory** [comment]
> You have modified or added at least one vcpkg.json where you should check the `license` field.

The software uses as custom license, so I added `license: null`.

**Comment by LilyWangLL** [comment]
Sorry I am late, I did a little change, just formatted the portfile.cmake and merged master, added "" for absolute path.

**Review by LilyWangLL — APPROVED** [review]

File: ports/random123/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+# Copy the headers that define this package to the install location.
+file(GLOB header_files 
+     "${SOURCE_PATH}/include/Random123/*.h"
+     "${SOURCE_PATH}/include/Random123/*.hpp") 
+file(COPY ${header_files}
+     DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}")
+file(COPY "${SOURCE_PATH}/include/Random123/conventional"
+     DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}")
+file(COPY "${SOURCE_PATH}/include/Random123/features"
+     DESTINATION "${CURRENT_PACKAGES_DIR}/include/${PORT}")
```
Comment (ras0219-msft) [inline ports/random123/portfile.cmake:22]: It looks like the header path is intended to be capitalized.

```suggestion
file(GLOB header_files
    "${SOURCE_PATH}/include/Random123/*.h"
    "${SOURCE_PATH}/include/Random123/*.hpp")
file(COPY ${header_files}
    "${SOURCE_PATH}/include/Random123/features"
    "${SOURCE_PATH}/include/Random123/conventional"
    DESTINATION "${CURRENT_PACKAGES_DIR}/include/Random123")
```

File: ports/random123/vcpkg.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "random123",
+  "version": "1.14.0",
+  "description": "Random123 is a library of 'counter-based' random number generators (CBRNGs), in which the Nth random number can be obtained by applying a stateless mixing function to N instead of the conventional approach of using N iterations of a stateful transformation.",
+  "homepage": "http://www.deshawresearch.com/resources_random123.html",
+  "license": null
```
Comment (ras0219-msft) [inline ports/random123/vcpkg.json:6]:
```suggestion
  "license": "BSD-3-Clause"
```

File: ports/random123/portfile.cmake [hunk 9]
```diff
@@ -0,0 +1,27 @@
+# Random123 - Header-only library
+
+set(VERSION 1.14.0)
+
+vcpkg_from_github(
+  OUT_SOURCE_PATH SOURCE_PATH
+  REPO "DEShawResearch/random123"
+  REF "v${VERSION}"
```
Comment (ras0219-msft) [inline ports/random123/portfile.cmake:8]:
```suggestion
  REF "v1.14.0"
```

File: ports/random123/portfile.cmake [hunk 10]
```diff
@@ -0,0 +1,27 @@
+# Random123 - Header-only library
+
+set(VERSION 1.14.0)
+
```
Comment (ras0219-msft) [inline ports/random123/portfile.cmake:4]:
```suggestion
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Pull request merged**